src/effects/CompressorInstance.cpp

```diff
@@ -0,0 +1,134 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  CompressorInstance.cpp
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#include "CompressorInstance.h"
+#include "processors/CompressorProcessor.h"
+
+CompressorInstance::CompressorInstance(const PerTrackEffect& effect)
+    : PerTrackEffect::Instance { effect }
+    , mCompressor { std::make_unique<CompressorProcessor>() }
+{
+}
+
+CompressorInstance::CompressorInstance(CompressorInstance&& other)
+    : PerTrackEffect::Instance { other }
+    , mCompressor { std::move(other.mCompressor) }
+    , mSlaves { std::move(other.mSlaves) }
+{
+}
+
+bool CompressorInstance::ProcessInitialize(
+   EffectSettings& settings, double sampleRate, ChannelNames chanMap)
+{
+   InstanceInit(settings, *mCompressor, GetAudioInCount(), sampleRate);
+   return true;
+}
+
+size_t CompressorInstance::ProcessBlock(
+   EffectSettings& settings, const float* const* inBlock,
+   float* const* outBlock, size_t blockLen)
+{
+   return InstanceProcess(settings, *mCompressor, inBlock, outBlock, blockLen);
+}
+
+bool CompressorInstance::RealtimeInitialize(EffectSettings&, double)
+{
+   SetBlockSize(512);
+   mSlaves.clear();
+   return true;
+}
+
+bool CompressorInstance::RealtimeSuspend()
+{
+   return true;
+}
+
+bool CompressorInstance::RealtimeResume()
+{
+   for (auto& slave : mSlaves)
+      // Neither block size nore sample rate or any other parameter has changed,
+      // so `Reinit()` should not reallocate memory.
+      slave.mCompressor->Reinit();
+   return true;
+}
+
+bool CompressorInstance::RealtimeAddProcessor(
+   EffectSettings& settings, EffectOutputs*, unsigned numChannels,
+   float sampleRate)
+{
+   mSlaves.emplace_back(std::move(mProcessor));
```

Comment:
This looks ... odd. I think there should be something in line with `mSlaves.push_back(CompressorInstance { mProcessor })`

Reply:
See `EffectReverb::Instance::RealtimeAddProcessor`